From linux-usb-owner@vger.kernel.org  Wed Sep  1 18:10:45 2010
Date: Tue, 31 Aug 2010 19:31:32 +0200
Message-ID: <AANLkTinezH-sqVzfzCZ8qjXrmaDXuwBRvRWYSN2Jw72+@mail.gmail.com>
Subject: USB: cdc-acm: Add pseudo modem without AT command capabilities
From: Philippe Corbes <philippe.corbes@gmail.com>
To: linux-usb@vger.kernel.org

From: Philippe Corbes <philippe.corbes@gmail.com>

cdc-acm.c : Manage pseudo-modem without AT commands capabilities
  Enable to drive electronic simple gadgets based on microcontrolers.
  The Interface descriptor is like this:
    bInterfaceClass         2 Communications
    bInterfaceSubClass      2 Abstract (modem)
    bInterfaceProtocol      0 None

Signed-off-by: Philippe Corbes <philippe.corbes@gmail.com>
Cc: stable <stable@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/cdc-acm.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -1615,6 +1615,10 @@ static const struct usb_device_id acm_id
 	.driver_info = NOT_A_MODEM,
        	},
 
+	/* control interfaces without any protocol set */
+	{ USB_INTERFACE_INFO(USB_CLASS_COMM, USB_CDC_SUBCLASS_ACM,
+		USB_CDC_PROTO_NONE) },
+
 	/* control interfaces with various AT-command sets */
 	{ USB_INTERFACE_INFO(USB_CLASS_COMM, USB_CDC_SUBCLASS_ACM,
 		USB_CDC_ACM_PROTO_AT_V25TER) },
